solc/CommandLineInterface.cpp

```diff
@@ -414,25 +416,28 @@ void CommandLineInterface::readInputFilesAndConfigureRemappings()
 			else
 			{
 				auto infile = boost::filesystem::path(path);
-				if (!boost::filesystem::exists(infile))
+				if (!boost::filesystem::exists(infile) || !boost::filesystem::is_regular_file(infile))
 				{
-					cerr << "Skipping non-existent input file \"" << infile << "\"" << endl;
-					continue;
-				}
+					//missing or irregular file, no direction to ignore missing files from command line
+					if(! m_args.count(g_argIgnoreMissingFiles)){
+						cerr << infile << " is not a valid file." << endl;
```

Comment:
Could you use two different error messages for the two cases?
Also, please format them like `Error: filename: No such file or directory`

Reply:
Will adopt that format and split back out into two messages.

Reply:
Fixed.